doc/isis/frr-isis-sonic-yang-model-hld.md

```diff
@@ -0,0 +1,652 @@
+ISIS Level Yang container is sonic-isis.yang.
+
+```
+        container ISIS_LEVEL {
+
+            list ISIS_LEVEL_LIST {
+
+                description
+                    "Configuration parameters related to a particular level within the
+                    IS-IS protocol instance";
+
+                key "instance level_number";
+
+                leaf instance {
+                    type string;
+                    description
+                        "The identifier for this instance of IS-IS. Area-tag";
+                }
+
+                leaf level_number {
+                    type stypes:level-number;
+                    description
+                        "IS-IS level number.";
+                }
+
+                leaf lsp_refresh_interval {
+                    type uint16;
+                    units "seconds";
+                    default "900";
+                    description
+                        "LSP refresh interval.";
+                }
+
+                leaf lsp_maximum_lifetime {
+                    type uint16 {
+                        range "350..65535";
+                    }
+                    units "seconds";
+                    must "(. >= ../lsp_refresh_interval + 300)" {
+                        error-message
+                            "lsp_maximum_lifetime must be at least 300 seconds greater than lsp_refresh_interval";
+                    }
+                    default "1200";
+                    description
+                        "Maximum LSP lifetime.";
+                }
+
+                leaf lsp_generation_interval {
+                    type uint16 {
+                        range "1..120";
+                    }
+                    units "seconds";
+                    must "(. < ../lsp_refresh_interval)" {
+                        error-message
+                            "lsp_generation_interval must be greater than lsp_refresh_interval";
+                    }
+                    default "30";
+                    description
+                        "Minimum time before an LSP retransmissions.";
+                }
+
+                leaf spf_minimum_interval {
+                    type uint16 {
+                        range "1..120";
+                    }
+                    units "seconds";
+                    default "1";
+                    description
+                        "Minimum time between consecutive SPFs.";
+                }
+
+            } // list ISIS_LEVEL_LIST
+
+        } // container ISIS_LEVEL
+```
+
+##### SONiC ISIS Interface
+ISIS Interface Yang container is sonic-isis.yang.
+
+```
+        container ISIS_INTERFACE {
+
+            list ISIS_INTERFACE_LIST {
+
+                description
+                    "Configuration parameters related to a particular interface within the
+                    IS-IS protocol instance";
+
+                key "instance ifname";
+
+                leaf instance {
+                    type string;
```

Comment:
Please use leafref to instance present in global table.

Reply:
Added a leafref to instance in either ISIS_LEVELS or ISIS_GLOBAL